force-app/main/default/lwc/ldsDeleteRecord/__tests__/ldsDeleteRecord.test.js

```diff
@@ -0,0 +1,109 @@
+
+            const element = createElement('c-lds-delete-record', {
+                is: LdsDeleteRecord
+            });
+            document.body.appendChild(element);
+            getAccountListAdapter.emit(mockGetAccountList);
+            return Promise.resolve().then(() => {
+                const nameEl = element.shadowRoot.querySelector(
+                    'lightning-layout-item'
+                );
+                expect(nameEl.textContent).toBe(ACCOUNT_NAME_RESULT);
+
+                const buttonEls = element.shadowRoot.querySelectorAll(
+                    'lightning-button-icon'
+                );
+                expect(buttonEls.length).toBe(2);
+                expect(buttonEls[0].dataset.recordid).toBe(ACCOUNT_ID_RESULT);
+            });
+        });
+
+        it('with no record', () => {
+            const element = createElement('c-lds-delete-record', {
+                is: LdsDeleteRecord
+            });
+            document.body.appendChild(element);
+            getAccountListAdapter.emit(mockGetAccountListNoRecords);
+            return Promise.resolve().then(() => {
+                const nameEl = element.shadowRoot.querySelector(
+                    'lightning-button-icon'
+                );
+                expect(nameEl).toBeNull();
+
+                const detailEls = element.shadowRoot.querySelectorAll(
+                    'lightning-button-icon'
+                );
+                expect(detailEls.length).toBe(0);
+            });
+        });
+    });
+
+    describe('getAccountList @wire error', () => {
+        it('shows error panel element', () => {
+            const element = createElement('c-apex-wire-method-to-function', {
+                is: LdsDeleteRecord
+            });
+            document.body.appendChild(element);
+            getAccountListAdapter.error();
+            return Promise.resolve().then(() => {
+                const errorPanelEl = element.shadowRoot.querySelector(
+                    'c-error-panel'
+                );
+                expect(errorPanelEl).not.toBeNull();
+            });
+        });
+    });
+
+    it('deletes the first entry of the account list on button click', () => {
+        const INPUT_PARAMETER = ['0011700000r5uciAAA'];
+
+        const element = createElement('c-lds-delete-record', {
+            is: LdsDeleteRecord
+        });
+        document.body.appendChild(element);
+
+        getAccountListAdapter.emit(mockGetAccountList);
+
+        // Return a promise to wait for any asynchronous DOM updates.
+        return Promise.resolve()
+            .then(() => {
+                const buttonEl = element.shadowRoot.querySelector(
+                    'lightning-button-icon'
+                );
+                buttonEl.click();
+            })
+            .then(() => {
+                expect(deleteRecord).toHaveBeenCalled();
+                expect(deleteRecord.mock.calls[0]).toEqual(INPUT_PARAMETER);
```

Comment:
I'd love to add another test after `deleteRecord` for the updated list. How should the call of `refreshApex` in conjunction with the Apex wire adapter be handled?

@trevor-bliss @kevinv11n